src/transport_performance/gtfs/report/report_utils.py

```diff
@@ -0,0 +1,183 @@
+    "stops": [
+        "wheelchair_boarding",
+        "location_type",
+        "parent_station",
+        "platform_code",
+    ],
+    "calendar_dates": [],
+    "calendar": [],
+    "trips": [
+        "trip_headsign",
+        "block_id",
+        "shape_id",
+        "wheelchair_accessible",
+    ],
+    "shapes": [],
+}
+
+
+class TemplateHTML:
+    """A class for inserting HTML string into a docstring.
+
+    Attributes
+    ----------
+    template : str
+        A string containing the HTML template.
+
+    Methods
+    -------
+    insert(placeholder: str, value: str, replace_multiple: bool = False)
+        Insert values into the HTML template
+    get_template()
+        Returns the template attribute
+
+    """
+
+    def __init__(self, path: Union[str, pathlib.Path]) -> None:
+        """Initialise the TemplateHTML object.
+
+        Parameters
+        ----------
+        path : Union[str, pathlib.Path]
+            The file path of the html template
+
+        Returns
+        -------
+        None
+
+        """
+        _handle_path_like(path, "path")
+        with open(path, "r", encoding="utf8") as f:
+            self.template = f.read()
+        return None
+
+    def insert(
+        self, placeholder: str, value: str, replace_multiple: bool = False
+    ) -> None:
+        """Insert values into the html template.
+
+        Parameters
+        ----------
+        placeholder : str
+            The placeholder name in the template. This is a string. In the
+            template it should be surrounded by square brackets.
+        value : str
+            The value to place in the placeholder
+            location.
+        replace_multiple : bool, optional
+            Whether or not to replace multiple placeholders that share the same
+            placeholder value, by default False
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValueError
+            A ValueError is raised if there are multiple instances of a
+            place-holder but 'replace_multiple' is not True
+
+        """
+        _type_defence(placeholder, "placeholder", str)
+        _type_defence(value, "value", str)
+        _type_defence(replace_multiple, "replace_multiple", bool)
+        occurences = len(self.template.split(f"[{placeholder}]")) - 1
+        if occurences > 1 and not replace_multiple:
+            raise ValueError(
+                "You have selected not to replace multiple"
+                "placeholders of the same value, however"
+                "placeholders occur more than once. \n"
+                "If you would like to allow this, set the"
+                "replace_multiple param to True"
+            )
+
+        self.template = self.template.replace(f"[{placeholder}]", value)
+
+    def get_template(self) -> str:
+        """Get the template attribute of the TemplateHTML object.
+
+        Returns
+        -------
+        str
+            The template attribute
+
+        """
```

Comment:
Looks like the only reason we need this is to play nicely with pytest fixtures? Would be worth putting this reason within the docstring and ensuring this method is denoted as internal.